### PR TITLE
Various small tweaks

### PR DESCRIPTION
**Pull request by gkoz**
I'm hoping to convince cargo to print a meaningful message instead of "unknown error" when the exit code is non-zero.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -30,53 +33,36 @@ mod version;
...
-            ::std::process::exit(1);
+        Err(config::Error::CommandLine(err)) => {
+            if err.fatal() {
+                return Err(Box::new(err));
+            }
+            else {
+                return Ok(())
+            }
         }
+        x => try!(x),
```
Comment (EPashkin) [inline src/main.rs]: IMHO separate Err(err) and Ok(c) will be clearer

**Comment by gkoz** [comment]
Yes, turns out it also was possible to make the `match` smaller.

**Comment by EPashkin** [comment]
`codegen-units` really speed up build.
Thanks.

**Comment by gkoz** [comment]
@homu r+

**Comment by gkoz** [comment]
@homu r+

File: src/main.rs [hunk 2]
```diff
@@ -30,53 +33,30 @@ mod version;
...
 
     let cfg = match config::Config::new() {
-        Ok(c) => c,
-        Err(config::Error::CommandLine(e)) => e.exit(),
-        Err(e) => {
-            println!("{}", e);
-            ::std::process::exit(1);
-        }
+        Ok(cfg) => cfg,
+        Err(config::Error::CommandLine(ref err)) if !err.fatal() => return Ok(()),
```
Comment (EPashkin) [inline src/main.rs]: Missed that `gir --help` (not fatal) don't print anything :(

**Pull request merged**